Sources/EdgeNetworkHandlers/NetworkResponseHandler.swift

```diff
@@ -88,39 +88,20 @@ class NetworkResponseHandler {
     ///   - requestId: request id associated with current response
     func processResponseOnError(jsonError: String, requestId: String) {
         guard let data = jsonError.data(using: .utf8) else { return }
-
-        // Attempt to decode as an `EdgeResponse` first
-        guard let edgeErrorResponse = try? JSONDecoder().decode(EdgeResponse.self, from: data) else {
-
-            // If decoding as an `EdgeResponse` fails, attempt to decode as a generic `EdgeEventError`
-            if let edgeErrorResponse = try? JSONDecoder().decode(EdgeEventError.self, from: data) {
-                Log.debug(label: LOG_TAG, "processResponseOnError - Processing server error response:\n \(jsonError), request id \(requestId)")
-                dispatchEventErrors(errorsArray: [edgeErrorResponse], requestId: requestId)
-            } else {
-                Log.warning(label: LOG_TAG,
-                            "processResponseOnError - The conversion to JSON failed for server error response: \(jsonError), request id \(requestId)")
-            }
-
-            return
-        }
-
         Log.debug(label: LOG_TAG, "processResponseOnError - Processing server error response:\n \(jsonError), request id \(requestId)")
 
-        // Note: if the Edge error doesn't have an eventIndex it means that this error is a generic request error,
-        // otherwise it is an event specific error. There can be multiple errors returned for the same event
-        if edgeErrorResponse.errors != nil {
+        if let edgeResponse = try? JSONDecoder().decode(EdgeResponse.self, from: data), edgeResponse.errors != nil {
```

Comment:
I think we should move edgeResponse.errors != nil inside the if or just let it fast fail in dispatchEventErrors. If we were able to decode to EdgeResponse we should not try to decode to EdgeEventError.
```
if let edgeResponse = try? JSONDecoder().decode(EdgeResponse.self, from: data) { ...
else if let edgeErrorResponse = try? JSONDecoder().decode(EdgeEventError.self, from: data) { ...
```